superduperdb/vector_search/postgres.py

```diff
@@ -0,0 +1,146 @@
+import json
+import typing as t
+import numpy
+from pgvector.psycopg import psycopg, register_vector
+
+
+from superduperdb.vector_search.base import BaseVectorSearcher, VectorItem
+
+
+class PostgresVectorSearcher(BaseVectorSearcher):
+    """
+    Implementation of a vector index using the ``pgvector`` library.
+    :param identifier: Unique string identifier of index
+    :param dimensions: Dimension of the vector embeddings in the Lance dataset
+    :param uri: connection string to postgres
+    :param h: ``torch.Tensor``
+    :param index: list of IDs
+    :param measure: measure to assess similarity
+    """
+
+    def __init__(
+        self,
+        identifier: str,
+        dimensions: int,
+        conninfo: str,
+        h: t.Optional[numpy.ndarray] = None,
+        index: t.Optional[t.List[str]] = None,
+        measure: t.Optional[str] = None,
+    ):
+        self.connection = psycopg.connect(conninfo=conninfo)
+        self.dimensions = dimensions
+        self.identifier = identifier
+        if measure == "l2" or not measure:
+            self.measure_query = "embedding <-> '%s'"
+        elif measure == "dot":
+            self.measure_query = "(embedding <#> '%s') * -1"
+        elif measure == "cosine":
+            self.measure_query = "1 - (embedding <=> '%s')"
+        else:
+            raise NotImplementedError("Unrecognized measure format")
```

Comment:
Please use `VectorIndexMeasureType`, and It would be better if it could be simply turned into a function

Reply:
done